### PR TITLE
🐛Make inventoryManager implement the InventoryManager interface

### PR DESCRIPTION
**Pull request by jaym**
We also don't expose public things on inventoryManager and instead expose a method on the interface

### CODE REVIEW

File: motor/inventory/manager.go [hunk 1]
```diff
@@ -65,7 +68,7 @@ func WithVault(v vault.Vault) Option {
 
 func WithCachedCredsResolver() Option {
 	return func(im *inventoryManager) error {
-		im.CredsResolver = credentials_resolver.New(im.GetVault(), false)
```
Comment (jaym) [inline motor/inventory/manager.go:68]: This branch has enableCaching always false. Not sure what's going on

Comment (chris-rock) [reply motor/inventory/manager.go:68]: That does not make any sense. But we should find occurrences of WithCachedCredsResolver and remove them for now

Comment (chris-rock) [reply motor/inventory/manager.go:68]: and set isCached to true here

**Review by chris-rock — APPROVED** [review]
Thank you @jaym

**Pull request merged**